src/main/java/core/basesyntax/Rectangle.java

```diff
@@ -0,0 +1,29 @@
+package core.basesyntax;
+
+public class Rectangle extends Figure {
+    private double length;
+    private double width;
+
+    public Rectangle(String color, double length, double width) {
+        super(color);
+        this.length = length;
+        this.width = width;
+    }
+
+    public String toString() {
+        return "Figure: rectangle, area: " + calculateArea() + " sq. units, "
+                + "length: " + length + " units, "
+                + "width: " + width + " units, "
+                + "color: " + getColor();
+    }
+
+    @Override
+    public double calculateArea() {
+        return length * width;
+    }
+
+    @Override
+    public void draw(Figure figure) {
+        System.out.println(figure);
+    }

```

Comment:
The `draw` method should not be implemented in the `Rectangle` class. Instead, consider using an interface with a single purpose for drawing, as the `draw()` and `calculateArea()` methods are conceptually different.